### PR TITLE
#185218938 A buyer should be able to checkout

### PR DESCRIPTION
**Pull request by nishimweprince**
## What does this PR do?

This PR enables a buyer to validate their order and perform checkout to proceed to payment.

## Description of Task to be completed?

+ Given a buyer is logged into their account. They should be able to checkout a product.
+ Given a buyer checks out a product, the order is validated and updates the inventory
+ Given a buy checks out successfully, they are redirected to the payment portal to finalize purchase

## How should this be manually tested?

- Clone the repository using `git clone https://github.com/atlp-rwanda/e-comm-team-techsmith-fn`

- Checkout to the branch using `git checkout ft-buyer-checkout-185218938`

- Install node packages that the application depends on: `npm i`

- Create a `.env` file in the root directory and add all environment variables required to run the application. The required variables are listed in the `.env.example` file available in root directory of the application.

- Start the application development server: `npm start`

- Check out the components documentation: `npm run storybook`

## Any background context you want to provide?

The checkout button will call the `GET Single Order: '/single/:orderId'` to obtain the order details, and it is the payment button that will call the `POST Checkout Order: '/:id/checkout'`. This is because the `checkout` endpoint automatically performs payment (this is how we built it on the backend).

## What are the relevant pivotal tracker/Trello stories?

+ [#185218938](https://www.pivotaltracker.com/story/show/185218938)

## Screenshots (if applicable)
Screenshot 2023-06-22 at 03 45 52
Screenshot 2023-06-22 at 03 49 09
Screenshot 2023-06-22 at 04 06 51
Screenshot 2023-06-22 at 04 18 43
Screenshot 2023-06-22 at 04 19 36

### CODE REVIEW

**Comment by mken1205** [comment]
You should put your commit message in the format that was given to us

**Review by Dyves1 — APPROVED** [review]

File: src/containers/Hero.jsx
Comment (cyubahiro367) [inline src/containers/Hero.jsx:1]: plz keep that text of new arrival

File: src/components/ValidateOrder.jsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    createOrder({
+      desiredQuantity: formData.quantity,
+      productId: formData.productId,
+      amount: formData.amount
+    });
+    if (isSuccess) {
+      dispatch(setOrderId(orderData.data.id));
+      navigate(`/order/${orderData.data.id}`)
+    }
+  };
```
Comment (niyontwali) [inline src/components/ValidateOrder.jsx:44]: Same issue to the one of Frank. You can't capture isSuccess in the submission function. Try to use `useEffect` for that please. Work together with frank to resolve that.

**Review by niyontwali — CHANGES_REQUESTED** [review]

File: vercel.json
Comment (cyubahiro367) [inline vercel.json:1]: unless we changed the host i think we dont need vercel now

**Review by cyubahiro367 — CHANGES_REQUESTED** [review]

File: vercel.json
Comment (nishimweprince) [reply vercel.json:1]: Sure @cyubahiro367 great catch!

**Comment by nishimweprince** [comment]
> You should put your commit message in the format that was given to us

Ken you will notice that our commit message template has been changed, this new one helps us to update the PivotalTracker status automatically. Refer to this line in our [README.md](https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/blob/develop/README.md?plain=1#L52) file.

File: src/App.jsx [hunk 2]
```diff
@@ -59,7 +60,7 @@ const App = () => {
           <Route element={<ProtectedRoutes />}>
             <Route path='/chat' element={<JoinChat />} />
             <Route path='/chat/room' element={<ChatPage />} />
-            <Route path='/dashboard/users' element={<AdminManageUserPage />} />
```
Comment (jcharles921) [inline src/App.jsx:62]: this` dashboard/users` should be protected , only the admin should access it

**Review by jcharles921 — APPROVED** [review]
Great work

File: src/components/ValidateOrder.jsx (hunk 1 above)
Comment (nishimweprince) [reply src/components/ValidateOrder.jsx:44]: Please find the following workaround for your review:

```jsx
  const handleSubmit = () => {
    createOrder({
      desiredQuantity: formData.quantity,
      productId: formData.productId,
      amount: formData.amount
    });
  };

  useEffect(() => {
    if (isSuccess) {
      dispatch(setOrderId(orderData.data.id));
    }
  }, [orderData]);
```

File: src/App.jsx (hunk 2 above)
Comment (nishimweprince) [reply src/App.jsx:62]: Yes, done!

```jsx
<Route element={<AdminRoutes />}>
      <Route path='/dashboard/users' element={<AdminManageUserPage />} />
</Route>
```

Screenshot 2023-06-22 at 11 51 01

**Review by niyontwali — APPROVED** [review]

**Review by cyubahiro367 — APPROVED** [review]

**Review by bergad37 — APPROVED** [review]
Great work!

**Review by irakevine — APPROVED** [review]
Great Work

**Pull request merged**